### PR TITLE
Prevent silently mixing translation domains in statistics

### PR DESCRIPTION
**Pull request by yvanzo**
# Problem

Following discussions around #3139, the aliasing of `l_statistics` to `l` doesn’t prevent `exp.l` to be used and not previously imported `lp` to be used. This led to translations to not be used because these were searched in the translation domain `server` instead of `statistics`.

# Solution

* Stop aliasing `l_statistics` to `l` and just use plain `l_statistics` and its variants
* Add a basic CI test to check that there is no explicit call to `l`, `exp.l`, `texp.l` and their variants in statistics code.

Besides, while updating the `musicbrainz-tests` images to `v-2024-01.1`:

* Remove the country code for Greek and Spanish, to follow the same changes that occurred in production.

# Testing

* [x] Using sample data (that has no statistics), open the page http://localhost:5000/statistics and check that the main paragraph is still translated (in French, and probably in other production languages too).
* [x] Copying an event from the main database, open the page http://localhost:5000/history and check that the strings are actually translated (in Italian, which has the best coverage).
* [x] Tested Flow and ESLint on modified JS files.

# Draft progress

* [x] Let the added CircleCI test pass

### CODE REVIEW

**Review by reosarevok — COMMENTED** [review]
For what it's worth, it seems that we force translators to retranslate a ton of lines here that are already in `musicbrainz_server`, including entity names and the like, and which we should possibly reconsider reusing. But for now, let's just stick with this change :)

File: root/statistics/Countries.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import * as React from 'react';
 import {CatalystContext} from '../context.mjs';
 import * as manifest from '../static/manifest.mjs';
 import EntityLink from '../static/scripts/common/components/EntityLink.js';
-import {l_statistics as l} from '../static/scripts/common/i18n/statistics.js';
+import {l_statistics} from '../static/scripts/common/i18n/statistics.js';
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply root/statistics/Countries.js:16]: Not necessarily but it isn’t currently in `providePluginConfig.mjs` and it isn’t used outside of `root/statistics/`.

File: root/statistics/Edits.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import * as React from 'react';
 import LinkSearchableEditType from '../components/LinkSearchableEditType.js';
 import {CatalystContext} from '../context.mjs';
 import {l as lMbServer} from '../static/scripts/common/i18n.js';
```
[the remark this answers is not in the document]
Comment (yvanzo) [reply root/statistics/Edits.js:14]: Exactly, the reason for this alias just changed, from not colliding with the other (now removed) alias, to making it intentional and passing the added test.

**Comment by yvanzo** [comment]
> we force translators to retranslate a ton of lines here that are already in musicbrainz_server

Yes there are a lot of duplicates between the translation components Server and Statistics, but the burden on translators is reduced through _translation propagation_ which is enabled for MusicBrainz components in Weblate: _Whether translation updates in other components will cause automatic translation in this one_. It is a good thing for Server / Statistics. Not sure whether it might be problematic for any other component.

But more generally, I agree with you that the translation domains/components should be revisited at some point, based on translators’ feedback.

File: root/statistics/Countries.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/statistics/Countries.js:16]: Isn't it part of the `GETTEXT_DOMAINS.forEach` set?

**Review by reosarevok — APPROVED** [review]
LGTMBDNT

**Pull request merged**